### DISCUSSION TITLE
Intermediate destinations

### DISCUSSION THREAD

**Discussion by Malakay-666** [organicmaps/organicmaps#7018] [Ideas]
Hi,
I came from Sailfish OS Pure Maps to Calyx OS with Organic maps. I was just wondering - could be possible to add certain feature?
My point is that I am missing "intermediate destinations". 
For example - I want to go from point "A" to point "B", app calculates some route. I don´t like the route, so I add "intermediate destinations", points "C", "D" and "E" and this way I can modify the route according my preferences. 
I noticed I can do this when I make long tap on some place on map and choose "Přidat zastávku" (in english I think it means "Add stop" ). But I think little better could be when this option will be available in routing settings (where paid roads are on/off, highways etc).

And one more thing - when I reach my destination, is there any option to use last route, but in opposite direction to get back to start through the same way?

Thanks for the help :)

**Comment by biodranik** [comment]
When a route is built, you can select any object on the map by a single or a long tap, or select any search result, and press "Add stop". So it is already possible to add up to 100 intermediate stops.

**Reply by Malakay-666** [reply]
Yeah, I noticed this. But it has one minor disadvantage - when I tap on some place and add it as a stop, it becomes a waypoint - I don´t need precisely this point, so the route could be unnecessarily complicated. That "list of stops" could be better, because you write only names of cities, so it will route you through that city, but not through specific place you long-tappped. My english sucks, I hope I explained it correctly, what i mean.

Maybe better explanation - when I long tap on some waypoint, its address is for example Street 123, XYZ City. So it leads me through this address, but I want to be directed only through XYZ City (perhaps fastest route through), I don´t want to go via that specific address. Do you know what I mean? :)

**Reply by Malakay-666** [reply]
![Bez názvu-2](https://github.com/organicmaps/organicmaps/assets/36851792/8af36e18-540c-4478-a952-8915846c2087)
I made an example of route, leading through Olomouc.
On the left side is route like Pure Maps on Sailfish OS calculated for me, when I inserted Olomouc as an intermediate destination (not specific address, just the name of the city). It chose non-paid highway around centre of Olomouc - the fastest way.
On the right side I made an example, when I see non-much detailed map of route and long tap Olomouc as a stop - I long tapped some address and now Organic maps routes me almost through the center of Olomouc.

**Reply by Malakay-666** [reply]
![Snimek_20231230_001 1](https://github.com/organicmaps/organicmaps/assets/36851792/5e24e4b8-3340-4289-8d1f-d3ad448ae7a8)
![Snimek_20231230_002 1](https://github.com/organicmaps/organicmaps/assets/36851792/47f640af-8bcd-498f-b1cf-ffd5ad9d07a4)

**Reply by Malakay-666** [reply]
On the upper image you can see I created route from Prague to Brno. But I added waypoints (just wrote names of all cities I want go through) and it created route according my preference (on second image)

**Reply by pastk** [reply]
- looks same as https://github.com/organicmaps/organicmaps/issues/2699
